glowbot/commands.py

```diff
@@ -181,16 +193,26 @@ async def claim(self, ctx: discord.ApplicationContext, hours: Option(
 
                         expiration = cur_expiration + timedelta(hours=grant_value)
 
-                    # Make sure all RCON grants are successful.
-                    result_dict = await self.client.grant_vip(player.player_name, player.steam_id_64, expiration.strftime("%Y-%m-%dT%H:%M:%S%z"))
-                    for rcon, result in result_dict.items():
-                        if result is False:
-                            self.logger.error(f'Problem assigning VIP in `claim` for \"{rcon}\": {result}')
-                            await ctx.respond(f'{ctx.author.mention}: There was a problem on one of the servers assigning your VIP.')
-                            return
+                    message = ""
+                    if cur_expiration >= datetime(year=2200, month=1, day=1, tzinfo=timezone.utc): 
+                        # non-expiring vip... converting seeding hours is pointless...
+                        message += "Your VIP does not expire... no need to convert seeding hours!"
+                    else:
+                        # Make sure all RCON grants are successful.
+                        result_dict = await self.client.grant_vip(player.player_name, player.steam_id_64, expiration.strftime("%Y-%m-%dT%H:%M:%S%z"))
+                        for rcon, result in result_dict.items():
+                            if result is False:
+                                self.logger.error(f'Problem assigning VIP in `claim` for \"{rcon}\": {result}')
+                                await ctx.respond(f'{ctx.author.mention}: There was a problem on one of the servers assigning your VIP.')
+                                return
+
+                        # !!! should only decrease banked seeding time if it is actually used...
+                        player.seeding_time_balance -= timedelta(hours=hours)
+
+                        message += f'{ctx.author.mention}: You\'ve added `{grant_value}` hour(s) to your VIP status.'
+                        message += f'\nYour VIP expiration date is <t:{int(expiration.timestamp())}:R>'
 
-                    message = f'{ctx.author.mention}: You\'ve added `{grant_value}` hour(s) to your VIP status.'
-                    message += f'\nYou have VIP until `{expiration}`'
+                    # common messages...
```

Comment:
Pedantic: I'd ask we remove the comments where the behavior is explained more-or-less by the above code.  I'm all for documenting these in PR's/issues however where they can appear in `git blame` and various other places.  And of course comments are welcome where either code is not clear and/or a process needs to be explained.